### PR TITLE
Feature/flower it2

### LINKED ISSUE #59: New for for Flower Shop
User Story 11, Parent Creation (x2)

As a visitor
When I visit the Parent Index page
Then I see a link to create a new Parent record, "New Parent"
When I click this link
Then I am taken to '/parents/new' where I  see a form for a new parent record
When I fill out the form with a new parent's attributes:
And I click the button "Create Parent" to submit the form
Then a `POST` request is sent to the '/parents' route,
a new parent record is created,
and I am redirected to the Parent Index page where I see the new Parent displayed.

### LINKED ISSUE #60: Update Flower Shop
User Story 12, Parent Update (x2)

As a visitor
When I visit a parent show page
Then I see a link to update the parent "Update Parent"
When I click the link "Update Parent"
Then I am taken to '/parents/:id/edit' where I  see a form to edit the parent's attributes:
When I fill out the form with updated information
And I click the button to submit the form
Then a `PATCH` request is sent to '/parents/:id',
the parent's info is updated,
and I am redirected to the Parent's Show page where I see the parent's updated info

### LINKED ISSUE #61: Flower shop flowers create
User Story 13, Parent Child Creation (x2)

As a visitor
When I visit a Parent Childs Index page
Then I see a link to add a new adoptable child for that parent "Create Child"
When I click the link
I am taken to '/parents/:parent_id/child_table_name/new' where I see a form to add a new adoptable child
When I fill in the form with the child's attributes:
And I click the button "Create Child"
Then a `POST` request is sent to '/parents/:parent_id/child_table_name',
a new child object/row is created for that parent,
and I am redirected to the Parent Childs Index page where I can see the new child listed

### LINKED ISSUE #62: Flowers update
User Story 14, Child Update (x2)

As a visitor
When I visit a Child Show page
Then I see a link to update that Child "Update Child"
When I click the link
I am taken to '/child_table_name/:id/edit' where I see a form to edit the child's attributes:
When I click the button to submit the form "Update Child"
Then a `PATCH` request is sent to '/child_table_name/:id',
the child's data is updated,
and I am redirected to the Child Show page where I see the Child's updated information

### LINKED ISSUE #64: update index with truthy values only
User Story 15, Child Index only shows `true` Records (x2)

As a visitor
When I visit the child index
Then I only see records where the boolean column is `true`

### LINKED ISSUE #65: list flowers in alphabetical order
User Story 16, Sort Parent's Children in Alphabetical Order by name (x2)

As a visitor
When I visit the Parent's children Index Page
Then I see a link to sort children in alphabetical order
When I click on the link
I'm taken back to the Parent's children Index Page where I see all of the parent's children in alphabetical order

### LINKED ISSUE #66: Add links to update on index page
User Story 17, Parent Update From Parent Index Page (x2)

As a visitor
When I visit the parent index page
Next to every parent, I see a link to edit that parent's info
When I click the link
I should be taken to that parents edit page where I can update its information just like in User Story 4

### LINKED ISSUE #67: Add Flowers edit link to all flowers
User Story 18, Child Update From Childs Index Page (x1)

As a visitor
When I visit the `child_table_name` index page or a parent `child_table_name` index page
Next to every child, I see a link to edit that child's info
When I click the link
I should be taken to that `child_table_name` edit page where I can update its information just like in User Story 11

### PR DESCRIPTION
**Pull request by Jaybraum**
## Changes

- Added edit, create, update & new functionality to flowers & flower shops
- Added testing for 100% coverage
- Added links to pages for user stories
- 

## Special notes

This merge should put our test coverage at 100%. Could be missing some page content tests but Looks like most of it is covered 

## Checklist

- [x] Tests written where required
- [x] All tests passing
- [x] SimpleCov at 100%

closes #62 
closes #65  
closes #59
closes #60
closes #61
closes #67 
closes #66 
closes #64

### CODE REVIEW

File: app/views/flower_shops/edit.html.erb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<h1><%= @flower_shop.name %></h1>
+
+<%= bootstrap_form_with url: "/flower_shops/#{@flower_shop.id}", method: :patch, local: true do |form| %>
+  <%= form.text_field :name, value: @flower_shop.name %><br/>
+  <%= form.form_group :same_day_delivery, label: { text: "Same Day Delivery?" } do %><br/>
+    <%= form.radio_button :same_day_delivery, true, checked: @flower_shop.same_day_delivery, label: "Yes", inline:true %>
+    <%= form.radio_button :same_day_delivery, false, checked: @flower_shop.same_day_delivery, label: "No", inline:true %>
```
Comment (mkrumholz) [inline app/views/flower_shops/edit.html.erb:7]: You'll want the `false` option to be checked (read: true) when same_day_delivery is false, so you're missing a `!` like so:

`checked: !@flower_shop.same_day_delivery`

File: app/views/flowers/edit.html.erb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<h1><%= @flower.species %></h1>
+
+<%= bootstrap_form_with url: "/flowers/#{@flower.id}", method: :patch, local: true do |form| %>
+  <%= form.text_field :species, value: @flower.species %><br/>
+  <%= form.form_group :in_stock, label: { text: "In Stock?" } do %><br/>
+    <%= form.radio_button :in_stock, true, checked: @flower.in_stock, label: "Yes", inline:true %>
+    <%= form.radio_button :in_stock, false, checked: @flower.in_stock, label: "No", inline:true %>
+  <% end %><br/>
```
Comment (mkrumholz) [inline app/views/flowers/edit.html.erb:8]: same deal here with the `!` missing

File: spec/features/flower_shops/edit_spec.rb [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    visit "/flower_shops/#{@mikes_flowers.id}/edit"
+
+    fill_in "Rating", with: 5
+    click_button 'Update'
+
+    expect(current_path).to eq("/flower_shops/#{@mikes_flowers.id}")
+    expect(page).to have_content('Mikes Flower Shop')
+    expect(page).to have_content('Shop Rating: 5')
+  end
+end
```
Comment (mkrumholz) [inline spec/features/flower_shops/edit_spec.rb:47]: It looks like you're missing a test for the ordering of your shops based on created time.

File: spec/models/flower_shop_spec.rb [hunk 4]
```diff
@@ -4,8 +4,6 @@
   describe 'relationships' do
     it { should have_many :flowers }
   end
-
-  describe 'instance methods' do
```
Comment (mkrumholz) [inline spec/models/flower_shop_spec.rb:8]: Did you mean to remove this `describe` block? github makes it hard to see whether you have the right # of `end` statements now, so I'm assuming this was on purpose, but it looks odd to me here.

File: spec/features/flower_shops/edit_spec.rb (hunk 3 above)
Comment (Jaybraum) [reply spec/features/flower_shops/edit_spec.rb:47]: Thats in the flower shops index spec!

**Pull request merged**